### PR TITLE
Use Ipopt_jll as the executable

### PR DESCRIPTION
**Pull request by odow**
This should be sufficient to use Ipopt_jll as the executable. It works for me locally on macOS.

### CODE REVIEW

**Comment by odow** [comment]
I haven't tested this with HSL, which might be the only issue. Although that should work if it's on your path

**Comment by ccoffrin** [comment]
Didn't seem to work on the linux system I use,

```
julia> include("pyomo.jl"); solve_opf("$(@__DIR__)/../data/pglib_opf_case5_pjm.m")
/usr/bin/ssh: symbol lookup error: /lib64/libk5crypto.so.3: undefined symbol: EVP_KDF_ctrl, version OPENSSL_1_1_1b
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
ERROR: LoadError: failed process: Process(setenv(...
...
Stacktrace:
  [1] pipeline_error
    @ ./process.jl:565 [inlined]
  [2] run(::Cmd; wait::Bool)
    @ Base ./process.jl:480
  [3] run
    @ ./process.jl:477 [inlined]
  [4] cached_git_clone(url::String; desired_commit::Nothing, clones_dir::String, verbose::Bool)
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/git_utils.jl:42
  [5] get_commit_sha(url::String, tree_hash::Base.SHA1; verbose::Bool)
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/git_utils.jl:57
  [6] get_commit_sha
    @ ~/.julia/packages/JLLPrefixes/pWvDL/src/git_utils.jl:56 [inlined]
  [7] get_addable_spec(name::String, version::VersionNumber; ctx::Pkg.Types.Context, verbose::Bool)
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/pkg_utils.jl:122
  [8] (::JLLPrefixes.var"#25#37"{Base.BinaryPlatforms.Platform, Bool, Vector{Pkg.Types.PackageSpec}, Dict{Pkg.Types.PackageSpec, Dict}, Vector{String}})()
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:123
  [9] activate(f::JLLPrefixes.var"#25#37"{Base.BinaryPlatforms.Platform, Bool, Vector{Pkg.Types.PackageSpec}, Dict{Pkg.Types.PackageSpec, Dict}, Vector{String}}, new_project::String)
    @ Pkg.API ~/bin/julia-1.9.0/share/julia/stdlib/v1.9/Pkg/src/API.jl:1847
 [10] #24
    @ ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:78 [inlined]
 [11] with_depot_path(f::JLLPrefixes.var"#24#36"{Base.BinaryPlatforms.Platform, Bool, Vector{Pkg.Types.PackageSpec}, String, Dict{Pkg.Types.PackageSpec, Dict}, Vector{String}}, new_path::String)
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/pkg_utils.jl:181
 [12] #23
    @ ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:78 [inlined]
 [13] with_no_pkg_handrails(f::JLLPrefixes.var"#23#35"{Base.BinaryPlatforms.Platform, String, Bool, Vector{Pkg.Types.PackageSpec}, String, Dict{Pkg.Types.PackageSpec, Dict}, Vector{String}})
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/pkg_utils.jl:32
 [14] collect_artifact_metas(dependencies::Vector{Pkg.Types.PackageSpec}; platform::Base.BinaryPlatforms.Platform, project_dir::String, pkg_depot::String, verbose::Bool)
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:78
 [15] collect_artifact_metas(dependencies::Vector{Pkg.Types.PackageSpec})
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:46
 [16] collect_artifact_paths(dependencies::Vector{Pkg.Types.PackageSpec}; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:211
 [17] collect_artifact_paths(dependencies::Vector{Pkg.Types.PackageSpec})
    @ JLLPrefixes ~/.julia/packages/JLLPrefixes/pWvDL/src/JLLPrefixes.jl:210
 [18] top-level scope
    @ ~/nlp-comp/rosetta-opf/variants/pyomo.jl:19
 [19] include(fname::String)
    @ Base.MainInclude ./client.jl:478
 [20] top-level scope
    @ REPL[1]:1
in expression starting at /vast/home/cjc/nlp-comp/rosetta-opf/variants/pyomo.jl:13
```

**Comment by odow** [comment]
Is it a standard system? Or are there some firewall issues? Does it have `git` installed?

**Comment by ccoffrin** [comment]
It does have git installed, but external access is tricky.  If there is a solution that locates the binary on the local file system without having to call out, that should work.

**Comment by odow** [comment]
Can you run `] dev https://github.com/JuliaBinaryWrappers/Ipopt_jll.jl` on the machine? How do you install other Julia packages?

**Comment by odow** [comment]
It looks like there are hooks to use a manual directory for git clones, so that you could download them locally, copy to the machine, and then update `set_git_clones_dir!` to point to that directory:

https://github.com/JuliaPackaging/JLLPrefixes.jl/blob/f1db43577625a94f374b2179a008bc52f489541e/src/JLLPrefixes.jl#L23-L27

https://github.com/JuliaPackaging/JLLPrefixes.jl/blob/f1db43577625a94f374b2179a008bc52f489541e/src/git_utils.jl#L31-L42

but I've opted just to hard-code the dependencies for now.

File: variants/pyomo.jl [hunk 1]
```diff
@@ -7,6 +7,33 @@
 import PowerModels
 import PythonCall
 
+# This block of code grabs all of the Ipopt_jll dependencies as copies them into
+# a single directory, so that we can pass /tmp/ipopt_jll/bin/ipopt to Pyomo.
+# It only needs to be run once.
+if !isdir("/tmp/ipopt_jll")
+    import Ipopt_jll, JLLPrefixes, Pkg
+    # For LANL's restricted machines, hard-code the dependencies that are needed.
+    import Ipopt_jll, METIS_jll, MUMPS_seq_jll, CompilerSupportLibraries_jll, OpenBLAS32_jll, ASL_jll
```
Comment (odow) [inline variants/pyomo.jl:16]: You'll need these installed. They should already be transitive dependencies, so they'll already be installed. Just need them as top-level packages.

**Pull request closed without merging**